### ISSUE TITLE
Nested stack API Gateway Lambda authorizer URI parameter incorrectly fails I3042:accountId check

### ISSUE THREAD

**Issue by Slooz** [aws-cloudformation/cfn-lint#3716]
### CloudFormation Lint Version

1.15.1

### What operating system are you using?

Mac

### Describe the bug

Continuation of #3683. #3683 fixed the issue when the URI is passed to a resource type of `AWS::ApiGateway::Authorizer`, but the issue persists when the URI is passed as a parameter for a resource type of `AWS::CloudFormation::Stack`.

cfn-lint expects the `:lambda:` segment of the stack API Gateway Lambda authorizer URI parameter to not be a hardcoded account ID, when the `:lambda:` segment should be hardcoded to `lambda`.

```
$ cfn-lint --format pretty --regions us-east-2 --include-checks I --configure-rule E3012:strict=True I3042:region=True I3042:accountId=True -- template.yaml
template.yaml
10:9:               I3042     ARN in Resource Stack contains hardcoded AccountId in ARN or incorrectly placed Pseudo Parameters
```

### Expected behavior

Stack API Gateway Lambda authorizer URI parameter doesn't fail I3042:accountId check

### Reproduction template

```yaml
AWSTemplateFormatVersion: 2010-09-09
Resources:
  Stack:
    Type: AWS::CloudFormation::Stack
    DeletionPolicy: Delete
    UpdateReplacePolicy: Delete
    Properties:
      TemplateURL: !Sub https://s3_bucket_name.s3.${AWS::Region}.amazonaws.com/template.yaml
      Parameters:
        AuthorizerUri: !Sub arn:${AWS::Partition}:apigateway:${AWS::Region}:lambda:path/2015-03-31/functions/arn:${AWS::Partition}:lambda:${AWS::Region}:${AWS::AccountId}:function:FunctionName/invocations
```
